pkg/ui/workspaces/cluster-ui/src/api/fetchData.ts

```diff
@@ -37,7 +37,10 @@ export function toArrayBuffer(encodedRequest: Uint8Array): ArrayBuffer {
  * TimeoutUnit ( Hour → "H", Minute → "M", Second → "S", Millisecond → "m" ),
  * e.g. "1M" (1 minute), default value "30S" (30 seconds);
  **/
-export const fetchData = <P extends ProtoBuilder<P>, T extends ProtoBuilder<T>>(
+export const fetchData = async <
+  P extends ProtoBuilder<P>,
+  T extends ProtoBuilder<T>,
+>(
```

Comment:
I'm a bit less educated re: `async`. Can you explain how adding `async` to this type doesn't break callsites? The return type stays the same?

Reply:
Missed this initially. From the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function):
```
The async function declaration creates a binding of a new async function to a given name. The await keyword is permitted within the function body, enabling asynchronous, promise-based behavior to be written in a cleaner style and avoiding the need to explicitly configure promise chains.
```

Ultimately it's to enable the `await` syntax to avoid having to use `.then` which can result in code that's easier to read.